kbfsblock/id.go

```diff
@@ -91,6 +104,37 @@ func MakeTemporaryID() (ID, error) {
 	return ID{h}, nil
 }
 
+// MakeRandomIDInRange generates a random block ID using a CSPRNG, distributing
+// the random variable over the interval [start, end), where the full range is
+// [0, MaxUint64). This corresponds to a normalized representation of the
+// range [kbfshash.RawDefaultHash{}, kbfshash.MaxDefaultHash).
+func MakeRandomIDInRange(start, end uint64) (ID, error) {
+	if start >= end {
+		return ID{}, errors.New("Invalid range for random ID")
+	}
+	rangeSize := float64(end - start)
+	randBuf := make([]byte, 8)
+	err := kbfscrypto.RandRead(randBuf)
+	if err != nil {
+		return ID{}, err
+	}
+	// Generate a random unsigned int. Endianness doesn't matter here because
+	// the bytes are random.
+	randUint := binary.BigEndian.Uint64(randBuf)
+	randFloat := float64(randUint) / float64(math.MaxUint64)
```

Comment:
Yeah floats between 0 and 1 might be a good idea.

http://lemire.me/blog/2017/02/28/how-many-floating-point-numbers-are-in-the-interval-01/